runtime/src/iree-amd-aie/aie_runtime/iree_aie_runtime.h

```diff
@@ -200,6 +200,20 @@ inline ::XAie_TxnOpcode txnToTxn(XAie_TxnOpcode t) {
 // mlir-air legacy
 enum class AIEArch : uint8_t { AIE1 = 1, AIE2 = 2 };
 
+/// Contains additional device config parameters that can't be retrieved from
+/// aie-rt for whatever reason. Make sure the parameters can't be retrieved in
+/// another way before adding new fields to this struct.
+struct AMDAIEDeviceConfig {
+  /// Currently, the max arbiter/msel is hidden inside aie-rt.
+  uint8_t streamSwitchCoreArbiterMax{0};
+  uint8_t streamSwitchCoreMSelMax{0};
+  uint8_t streamSwitchMemTileArbiterMax{0};
+  uint8_t streamSwitchMemTileMSelMax{0};
+  uint8_t streamSwitchShimArbiterMax{0};
+  uint8_t streamSwitchShimMSelMax{0};
+  AMDAIEDeviceConfig() = default;
+};
```

Comment:
This is basically a nit/style thing but if there's no plan for anyone elsewhere to use this struct _as a struct_ (like grouped together) then it should be "flattened" into fields on `AMDAIEDeviceModel `. Or (if you think expanding the constructor args is tedious) then it should be at least moved into the definition of `AMDAIEDeviceModel ` like this

```
struct AMDAIEDeviceModel {
  ...
struct StreamSwitchPacketSwitchConfig {
  /// Currently, the max arbiter/msel is hidden inside aie-rt.
  uint8_t streamSwitchCoreArbiterMax{0};
  uint8_t streamSwitchCoreMSelMax{0};
  uint8_t streamSwitchMemTileArbiterMax{0};
  uint8_t streamSwitchMemTileMSelMax{0};
  uint8_t streamSwitchShimArbiterMax{0};
  uint8_t streamSwitchShimMSelMax{0};
  AMDAIEDeviceConfig() = default;
  } packetSwitchConfig;
```

Also if you do `const StreamSwitchPacketSwitchConfig&& packetSwitchConfig` as the arg you'll get the `std::move`s for free I think.

Reply:
> Or (if you think expanding the constructor args is tedious)

Yes, this is exactly the reason. A separate struct will be more versatile towards possibly future parameters.

> then it should be at least moved into the definition of AMDAIEDeviceModel

Sure, I can do that.

> Also if you do const StreamSwitchPacketSwitchConfig&& packetSwitchConfig as the arg you'll get the std::moves for free I think.

That will incur a copy then. I could make it `StreamSwitchPacketSwitchConfig&&`, but that would still need the move on the caller/user side and inside the constructor to avoid a copy. I prefer the pass by value here as it is more versatile towards move by value/move and towards adding a default value without needing multiple overloads.